Gaming The Syllabus/Assets/Scripts/DialogueBoxUIManager.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+
+/// <summary>
+/// Singleton UI manager that specifically manages the Dialogue Box within the UI.
+/// </summary>
+public class DialogueBoxUIManager : Singleton<DialogueBoxUIManager>
+{
+    // Start is called before the first frame update
+    [SerializeField] private TextMeshProUGUI dialogueBoxText;
+    private static LinkedList<string> dialogueStrings = new LinkedList<string>();
+
+    private void Awake()
+    {
+        InitializeSingleton();
+    }
+
+    /// <summary>
+    /// Adds a string (strToAdd) to the dialogue box within the UI.
+    /// </summary>
+    /// <param name="strToAdd">The string to be added to the dialogue box</param>
+    public static void addStringToDialogueBox(string strToAdd) {
+        if (dialogueStrings.Count >= 5) {
+            dialogueStrings.RemoveFirst();
+        }
+        dialogueStrings.AddLast(strToAdd);
+        _instance.UpdateDialogueBox();
+    }
+
+    /// <summary>
```

Comment:
Feel free to leave this if you want obviously but just in regards to the XML commenting (not related to our XML file stuff, it's just the way C# docs are usually done I think) I usually don't do the big comments for private methods in most cases. Don't feel pressured to do them in general btw, but if you wanna do them on public stuff I definitely personally like it so I won't complain